src/Microsoft.Azure.SignalR.Common/ServiceMessages/RuntimeServicePingMessages.cs

```diff
@@ -62,18 +62,17 @@ public static bool TryGetStatus(this ServicePingMessage ping, out bool isActive)
             return true;
         }
 
-        public static bool TryGetServerIds(this ServicePingMessage ping, out HashSet<string> serverIds, out long updatedTime)
+        public static bool TryGetServerIds(this ServicePingMessage ping, out string serverIds, out long updatedTime)
         {
             // servers ping format: { "servers", "1234567890:server1;server2;server3" }
             if (!TryGetValue(ping, ServersKey, out var value) || string.IsNullOrEmpty(value)
                 || !long.TryParse(value.Substring(0, value.IndexOf(":")), out updatedTime))
             {
-                serverIds = null;
+                serverIds = string.Empty;
                 updatedTime = DateTime.MinValue.Ticks;
                 return false;
             }
-            var servers = value.Substring(value.IndexOf(":") + 1);
-            serverIds = new HashSet<string>(servers.Split(new char[] { ServerListSeparator }, StringSplitOptions.RemoveEmptyEntries));
+            serverIds = value.Substring(value.IndexOf(":") + 1);
```

Comment:
1. `value.IndexOf` is calculated twice
2. if `serverIds` is empty for every endpoint, it is a match?

Reply:
For 2, no, empty will be ignored. Empty is invalid value. Cause if server side could receive the servers ping, it should include itself serverId at least.